[chore] Change processor config tests to unmarshal config only for that component. (part 1)

kovrus opened the pull request:
**Description:** <Describe what has changed.>
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->

Change processor config tests to unmarshal config for that component.

- [x] attributesprocessor
- [x] cumulativetodeltaprocessor
- [x] deltatorateprocessor
- [x] filterprocessor
- [x] groupbyattrsprocessor
- [x] groupbytraceprocessor

**Link to tracking Issue:** <Issue number if applicable>

Updates https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13224

kovrus committed:
[chore] change processor config tests to unmarshal config only for that component.

Updates https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13224

- attributesprocessor
- cumulativetodeltaprocessor
- deltatorateprocessor
- filterprocessor
- groupbyattrsprocessor
- groupbytraceprocessor

kovrus commented:
Some checks fail coz of https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/13325

bogdandrutu commented on processor/cumulativetodeltaprocessor/factory_test.go:
```diff
@@ -44,49 +46,42 @@ func TestCreateDefaultConfig(t *testing.T) {
-		factories.Processors[typeStr] = factory
-		config, err := servicetest.LoadConfigAndValidate(filepath.Join("testdata", test.configName), factories)
-		assert.NoError(t, err)
+			var id string
+			parts := strings.Split(k, "/")
+			if len(parts) > 1 {
+				id = parts[1]
+			}
+			sub, err := cm.Sub(config.NewComponentIDWithName(typeStr, id).String())
```
Use ComponentID.UnmarshalText()

bogdandrutu commented on processor/cumulativetodeltaprocessor/config_test.go:
```diff
@@ -71,59 +63,62 @@ func TestLoadingFullConfig(t *testing.T) {
-				assert.NotNil(t, config)
-				assert.NoError(t, err)
+			if tt.expected != nil {
+				assert.NoError(t, cfg.Validate())
+				assert.Equal(t, tt.expected, cfg)
 			} else {
-				assert.EqualError(t, err, fmt.Sprintf("processor %q has invalid configuration: %s", typeStr, test.errorMessage))
+				assert.EqualError(t, cfg.Validate(), tt.errorMessage)
```
I like this way more (that you also use in a different test):
```golang
if tt.expected == nil {
	assert.EqualError(t, cfg.Validate(), tt.errorMessage)
	return
}
assert.NoError(t, cfg.Validate())
assert.Equal(t, tt.expected, cfg)
```

bogdandrutu commented on processor/deltatorateprocessor/config_test.go:
```diff
@@ -15,83 +15,58 @@
-		expCfg     *Config
+		id           config.ComponentID
+		expected     config.Processor
+		errorMessage string
```
`errorMessage` not configured, remove and code associated with it.

kovrus replied:
It is provided in one of the test cases https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/13298/files#diff-8ecc2408749d30e707b9620e602b6cfceb1cfc30867e9f7eaaaf1f14f46f1d64R48

bogdandrutu commented on processor/deltatorateprocessor/factory_test.go:
```diff
@@ -44,49 +46,42 @@ func TestCreateDefaultConfig(t *testing.T) {
-		factories.Processors[typeStr] = factory
-		config, err := servicetest.LoadConfigAndValidate(filepath.Join("testdata", test.configName), factories)
-		assert.NoError(t, err)
+			var id string
+			parts := strings.Split(k, "/")
+			if len(parts) > 1 {
+				id = parts[1]
+			}
+			sub, err := cm.Sub(config.NewComponentIDWithName(typeStr, id).String())
```
Same use `ComponentID.UnmarshalText`. But not sure you need to do that, just simply `sub, err := cm.Sub(k)`

kovrus replied:
`sub, err := cm.Sub(k)` that is enough

bogdandrutu commented on processor/filterprocessor/factory_test.go:
```diff
@@ -80,24 +83,39 @@ func TestCreateProcessors(t *testing.T) {
-				if strings.Contains(test.configName, "traces") {
-					assert.Equal(t, test.succeed, tp != nil)
-					assert.Equal(t, test.succeed, tErr == nil)
+				var id string
+				parts := strings.Split(k, "/")
+				if len(parts) > 1 {
+					id = parts[1]
+				}
```
Same comment about the split.

kovrus committed:
[chore] Address comments.

kovrus committed:
[chore] run "make gotidy"

build-and-test / checks (pull_request) Failing after 3m — checks

Following instructions:
```
go.mod/go.sum deps changes detected, please run "make gotidy" and commit the changes in this PR.
```
